packages/golden-layout/src/utils/EventUtils.ts

```diff
@@ -1,13 +1,12 @@
-import { useEffect, useLayoutEffect, useRef } from 'react';
+import { useRef } from 'react';
 import EventEmitter from './EventEmitter';
 
 type AsArray<P> = P extends unknown[] ? P : [P];
 
-export type EventHandlerFunction<P = []> = (...parameters: AsArray<P>) => void;
 export type EventListenerRemover = () => void;
 export type EventListenFunction<TParameters = []> = (
   eventEmitter: EventEmitter,
-  handler: EventHandlerFunction<TParameters>
+  handler: (...params: AsArray<TParameters>) => void
```

Comment:
Just going to comment on this now. I changed this so the type errors are more friendly. With the type alias, you end up with something like `Argument '(id: int) => void' is not assignable to EventHandlerFunction<[id: string]>`.

By removing that type alias, you get a much cleaner error like `Argument '(id: int) => void' is not assignable to '(id: string) => void'`

Reply:
I find the first half of the error clearer, but I guess the last half with the specific incompatible param is the same. First half is just clear it's a function and what the params are. Not that `EventHandlerFunction` isn't fairly obvious it's a function, just less so than `(params) => void`

What difference would this make for defining the handler functions? You still just call `makeEventFunctions` the same way as previously.

I can put the alias back, I just found the error more direct without it

Reply:
Ehh it's in 3 places and I think `EventUtils` will be read/used much less frequently than adding new listeners and calling them will be. But I can change it back